### ISSUE TITLE
How to only read/write new logs from a specific pod using `ReadNamespacedPodLogWithHttpMessagesAsync`

### ISSUE THREAD

**Issue by snehapar9** [kubernetes-client/csharp#1491]
**Describe the bug**

ReadNamespacedPodLogWithHttpMessagesAsync reads all the logs (including historical ones) from a specific pod. I cannot find a good way to only read the recent logs that have not been read/written before in the previous iterations.
I tried the following but none of them actually fix this. 
1.Watch mechanism is not supported by ReadNamespacedPodLogWithHttpMessagesAsync so we are not able to detect any new logs using watch and only write them. Also added a watch mechanism to write only if there was an event detected, but this would work only if a pod was added/deleted/modified/errored in the namespace, any change in the log would not trigger an event, so this is not a viable solution.
2.Seek is not supported, so we cannot move the pointer to the end of the old logs and just write the new ones.
3.Experimented with the tailLines argument but it's not clear how to set the number of lines in this argument without doing extra read operations. Does not seem like a good solution to me. 

**Kubernetes C# SDK Client Version**
e.g. `9.0.1`

**Server Kubernetes Version**
e.g. `1.22.3`

**Dotnet Runtime Version**
e.g. net6

**To Reproduce**

**Expected behavior**

Capability to only read the recent logs that have not already been read from the pod.

Example - the following code reads all the logs (including historical ones) every time. The expectation is to read only the logs that have not been read in the previous iterations without having to hard code `tailLines` argument because this can risk losing some logs.

```
 while (!context.CancellationToken.IsCancellationRequested)
 {
     var stream = await _kubernetesClient.CoreV1.ReadNamespacedPodLogWithHttpMessagesAsync(pod.Metadata.Name, _kubernetesConfig.Namespace, container: container.Name);
     var reader = new StreamReader(stream.Body);
     while (!reader.EndOfStream)
     {
          //Write to console
     }

     await Task.Delay(1000);
 }

```

**KubeConfig**
If applicable, add a KubeConfig file with secrets redacted.

**Where do you run your app with Kubernetes SDK (please complete the following information):**
 - OS: [e.g. Linux]
 - Environment [e.g. container]
 - Cloud [e.g. Azure]

**Additional context**

**Comment by tg123** [comment]
check out example of https://github.com/kubernetes-client/csharp/blob/master/examples/logs/Logs.cs
it is `kubectl log -f`

you will get the steam of log output, thus, everything from the stream is new

**Comment by snehapar9** [comment]
Thanks @tg123! I tried giving this a shot by setting the follow flag to `true` but this does not seem to stream logs if I have it running inside a while loop till the cancellation token is requested. Are there any other alternatives?

**Comment by tg123** [comment]
here is how to test the example

(make sure there is one pod only)
```
kubectl run ping --image=busybox --command=true ping localhost
```

![image](https://github.com/kubernetes-client/csharp/assets/170430/8a2fd052-9a12-4a96-bba1-76c076fb945e)

**Comment by snehapar9** [comment]
Thank you @tg123! I was able to get this to work. 

However, seems like the cancellation token is not respected? I have a scenario where I'm constantly writing to the pod and simultaneously reading from it, the reader continues to read logs till the end of stream if after requesting a cancellation token.
I have already tried passing the cancellation token to the `ReadLineAsync` method but that does not seem to do the trick.
